### PR TITLE
Night mode

### PR DESCRIPTION
**Pull request by mltbnz**
## Night Mode
* adds Light and DarkTheme
* stores selected Theme in UserDefaults
* adds switch in settings to set theme

### Before creating the PR
- [x] Update CHANGELOG

### CODE REVIEW

**Comment by mltbnz** [comment]
Did someone already create a custom map-style for a dark mode?

**Comment by lennet** [comment]
> Did someone already create a custom map-style for a dark mode?

I think we have to wait until the sdk provides a public api for dark mode or  a way to style maps locally. 

The problem ist that custom styles for mapViews requires a webservices that provides the svg

**Comment by zutrinken** [comment]
> Did someone already create a custom map-style for a dark mode?

@mltbnz exzellent question!

@cbalster and I did some research on custom map tiles for Android based on OSM. The option is to host an own tile server, which might be quite expensive, or use a paid service. There are [Mapbox](https://www.mapbox.com) and [Thunderforest](https://www.thunderforest.com). The second could be an option in the future - they look quite open source friendly. Maybe there can be some cooperation, but I haven't reached out yet. That might be a job for @stephanlindauer with a better understanding of our traffic and approximately needed data package.

**Comment by mltbnz** [comment]
Seems like it is possible with a `MKTileOverlay` that points to a tile server. Would it be an option to use a already finished dark theme tile?

**Comment by mltbnz** [comment]
Found one here https://wiki.openstreetmap.org/wiki/Tile_servers but it doesn't not look really nice I think
Screenshot 2019-04-11 at 13 14 35

**Comment by headione** [comment]
Very cool @mltbnz . Can you please add it also to the CHANGELOG.md? https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md#added

https://keepachangelog.com/en/1.0.0/

Thanks :)

**Comment by mltbnz** [comment]
Great you took a look at the draft. The cell hightlightColor is one of the things on my list. Also changing the textColor of the overlay. I will also make the navigationBar non transparent. Another thing is the color of the chatBubble that is not handled and also I think I will create a seperate cell for each switch cell in settings since sometimes it seems to loose the reference to its selector I guess. Didn’t had the time to look into it properly. Other then that I am pretty happy with the result. The very dark map works kind of good

File: CriticalMass/AppController.swift [hunk 1]
```diff
@@ -46,17 +48,32 @@ class AppController {
...
-            .init(representation: .icon(UIImage(named: "Settings")!, accessibilityLabel: NSLocalizedString("settings.title", comment: "")), action: .navigation(viewController: getSettingsViewController)),
+            .init(representation: NavigationOverlayItem.Representation(button: rootViewController.followMeButton),
+                  action: .none),
+            .init(representation: NavigationOverlayItem.Representation(button: chatNavigationButtonController.button),
+                  action: .navigation(viewController: getSocialViewController)),
+            .init(representation: NavigationOverlayItem.Representation(button: kniggeButton),
+                  action: .navigation(viewController: getRulesViewController)),
+            .init(representation: NavigationOverlayItem.Representation(button: settingsButton),
+                  action: .navigation(viewController: getSettingsViewController)),
         ])
```
Comment (mltbnz) [inline CriticalMass/AppController.swift]: Updated these to be buttons also for accessability

**Comment by mltbnz** [comment]
There is still an issue with the night mode switch that sometimes is not switching the theme... Working on that. But the rest can be reviewed

**Comment by mltbnz** [comment]
settings switch cell issue should be fixed

**Comment by mltbnz** [comment]
So for top 1: Seems out of my reach since it is a 3rd party. Could not manipulate it so far.
top 2: Did you check out the latest code? Think I fixed it this morning :)
top 3 and 4 are fixed. Thanks

File: CriticalMass/Colors.swift [hunk 2]
```diff
@@ -68,10 +64,26 @@ extension UIColor {
...
+
     static var gray200: UIColor {
         return UIColor(red: 38 / 255.0, green: 38 / 255.0, blue: 38 / 255.0, alpha: 1)
     }
 
+    static var gray300: UIColor {
+        return UIColor(red: 69.0 / 255.0, green: 71.0 / 255.0, blue: 77.0 / 255.0, alpha: 1.0)
+    }
+
+    static var gray400: UIColor {
```
[the remark this answers is not in the document]
Comment (lennet) [reply CriticalMass/Colors.swift]: they might no bet up to date anymore. I started with this review before you pushed your latest changes

File: CriticalMass/ChatNavigationButton.swift [hunk 3]
```diff
@@ -24,9 +32,7 @@ class ChatNavigationButton: CustomButton {
     init() {
         super.init(frame: .zero)
         setImage(UIImage(named: "Chat")!, for: .normal)
-        tintColor = .navigationOverlayForeground
         adjustsImageWhenHighlighted = false
-        highlightedTintColor = UIColor.navigationOverlayForeground.withAlphaComponent(0.4)
```
Comment (lennet) [inline CriticalMass/ChatNavigationButton.swift]: The button doesn't update its color anymore for the pressed state

File: CriticalMass/MapViewController.swift [hunk 4]
```diff
@@ -58,18 +62,33 @@ class MapViewController: UIViewController, MKMapViewDelegate {
...
         configureMapView()
         condfigureGPSDisabledOverlayView()
     }
 
+    override func loadView() {
+        view = MKMapView(frame: .zero)
+    }
+
+    deinit {
+        NotificationCenter.default.removeObserver(self, name: NSNotification.themeDidChange, object: nil)
```
Comment (lennet) [inline CriticalMass/MapViewController.swift]: We don't need to unregister Notifications anymore

File: CriticalMass/MapViewController.swift [hunk 5]
```diff
@@ -79,6 +98,7 @@ class MapViewController: UIViewController, MKMapViewDelegate {
         NotificationCenter.default.addObserver(self, selector: #selector(positionsDidChange(notification:)), name: NSNotification.Name("positionOthersChanged"), object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(didReceiveInitialLocation(notification:)), name: NSNotification.Name("initialGpsDataReceived"), object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(updateGPSDisabledOverlayVisibility), name: NSNotification.Name("gpsStateChanged"), object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(themeDidChange), name: NSNotification.themeDidChange, object: nil)
```
Comment (lennet) [inline CriticalMass/MapViewController.swift]: Notification instead of NSNotification

File: CriticalMass/MapViewController.swift [hunk 6]
```diff
@@ -113,14 +133,18 @@ class MapViewController: UIViewController, MKMapViewDelegate {
...
-    public lazy var followMeButton: UserTrackingButton = {
-        let button = UserTrackingButton(mapView: mapView)
-        button.tintColor = .navigationOverlayForeground
-        return button
-    }()
-
     // MARK: Notifications
 
+    @objc private func themeDidChange() {
+        let theme = ThemeController().currentTheme
```
Comment (lennet) [inline CriticalMass/MapViewController.swift]: Shouldn't we inject the ThemeController or maybe pass the current theme as the notifications object instead of recreating a ThemeController

File: CriticalMass/NibProviding.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+//
+
+import UIKit
+
+protocol NibProviding {
+    static var typeName: String { get }
+    static var nib: UINib { get }
+}
+
+extension NibProviding where Self: NSObject {
```
Comment (lennet) [inline CriticalMass/NibProviding.swift]: Looks great! Maybe we could even add an extension for tableView that does the registration for a NibProviding

File: CriticalMass/TextFieldWithInsets.swift [hunk 8]
```diff
@@ -7,9 +7,64 @@
...
+        didSet {
+            placeholderColor = placeholderTextColor
+        }
+    }
+
+    // sets the backgroundColor when input begins
+    override func becomeFirstResponder() -> Bool {
+        let didBecomeFirstResponder = super.becomeFirstResponder()
+        if didBecomeFirstResponder {
+            setEditorBackgroundColor(to: textFieldBackgroundColor)
```
Comment (lennet) [inline CriticalMass/TextFieldWithInsets.swift]: Why is that needed? I don't see any difference in the backgroundColor after becoming or resigning first responder

File: CriticalMass/UIWindow+Appearance.swift [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+//  UIWindow+Appearance.swift
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 15.04.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import UIKit
+
+extension NSNotification.Name {
```
Comment (lennet) [inline CriticalMass/UIWindow+Appearance.swift]: Notification

File: CriticalMass/UIWindow+Appearance.swift [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    static let CMAppearanceWillRefreshWindow = NSNotification.Name(rawValue: "CMAppearanceWillRefreshWindowNotification")
+    static let CMAppearanceDidRefreshWindow = NSNotification.Name(rawValue: "CMAppearanceDidRefreshWindowNotification")
+}
+
+extension UIWindow {
+    @nonobjc private func _refreshAppearance() {
```
Comment (lennet) [inline CriticalMass/UIWindow+Appearance.swift]: Why do we have to remove and add all views in a Window?

Comment (mltbnz) [reply CriticalMass/UIWindow+Appearance.swift]: Otherwise changing the theme does not affect the window immediately. And to do that we have to build the view hierarchy again

File: CriticalMass/TextFieldWithInsets.swift (hunk 8 above)
Comment (mltbnz) [reply CriticalMass/TextFieldWithInsets.swift]: This is a workaround since otherwise activating the textfield always changed the backgroundColor of it when it became first responder. Took me some time but that was the only solution I found. I could comment it for better understanding why this code is there or maybe you know another option.

File: CriticalMass/NibProviding.swift (hunk 7 above)
Comment (mltbnz) [reply CriticalMass/NibProviding.swift]: yes. we could do something like that I think. But I would suggest we add that in another PR and then change it in the whole codebase

File: CriticalMass/MapViewController.swift (hunk 6 above)
Comment (mltbnz) [reply CriticalMass/MapViewController.swift]: Makes sense to inject it via from the AppController I think 👍

File: CriticalMass/AppDelegate.swift [hunk 11]
```diff
@@ -35,6 +35,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         window?.rootViewController = appController.rootViewController
         window?.makeKeyAndVisible()
 
+        UISearchBar.appearance() // ??
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/AppDelegate.swift]: Think not. Don't know why it was there anyways

File: CriticalMass/ThemeStore.swift [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    init(defaults: UserDefaults = .standard) {
+        self.defaults = defaults
+    }
+
+    /// Save Theme to UserDefaults
+    ///
+    /// - Parameter themeSelection: The Theme that will be saved.
+    func save(_ themeSelection: Theme) {
+        defaults.set(themeSelection.rawValue, forKey: defaultsKey)
+        defaults.synchronize()
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeStore.swift]: > Waits for any pending asynchronous updates to the defaults database and returns; this method is unnecessary and shouldn't be used.

Totally right

File: CriticalMass/Colors.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/Colors.swift]: I am sure which colors are not needed and I thought it is easier to use the ones from Zeplin. I can go over the colors and remove unused ones

[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/Colors.swift]: makes sense to do that in this PR maybe

File: CriticalMass/ThemeDefining.swift [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+    var refreshControlColor: UIColor {
+        return .black
+    }
+
+    var chatMessageInputTextViewBackgroundColor: UIColor {
+        return .gray500
+    }
+
+    var navigationOverlaySeperatorColor: UIColor {
+        return UIColor(red: 0.86, green: 0.88, blue: 0.85, alpha: 1.00)
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeDefining.swift]: I would say light and dark theme are different from the UIColor extension. The value types are representing the selection and just referencing the colors from the extension

File: CriticalMass/SettingsViewController.swift [hunk 14]
```diff
@@ -149,15 +120,33 @@ class SettingsViewController: UITableViewController {
...
+            guard let gpsSwitchCell = tableView.dequeueReusableCell(withIdentifier: String(describing: section.cellClass), for: indexPath) as? SettingsSwitchTableViewCell else {
+                fatalError("Should be a SettingsSwitchCell")
+            }
+            let isGPSEnabled = Preferences.gpsEnabled
+            let gpsHandler: SettingsSwitchHandler = { [unowned self] settingsSwitch in
+                self.gpsCellAction(settingsSwitch)
+            }
+            gpsSwitchCell.configure(isOn: isGPSEnabled, handler: gpsHandler)
+            cell = gpsSwitchCell
+        case .darkMode:
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/SettingsViewController.swift]: good idea

File: CriticalMass/SettingsViewController.swift [hunk 15]
```diff
@@ -149,15 +120,33 @@ class SettingsViewController: UITableViewController {
...
+                fatalError("Should be a SettingsSwitchCell")
+            }
+            let isNightModeEnabled = themeController.currentTheme == .dark ? true : false
+            let nightModeHandler: (UISwitch) -> Void = { [unowned self] settingsSwitch in
+                self.darkModeCellAction(settingsSwitch)
+            }
+            nightModeSwitchCell.configure(isOn: isNightModeEnabled, handler: nightModeHandler)
+            cell = nightModeSwitchCell
+        default:
+            cell = tableView.dequeueReusableCell(withIdentifier: String(describing: section.cellClass), for: indexPath)
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/SettingsViewController.swift]: I remove the default case and add explicilty add `.info` and `.github`

File: CriticalMass/ThemeController.swift [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    private func loadTheme() -> Theme {
+        guard let theme = store.load() else {
+            return .light
+        }
+        return theme
+    }
+
+    func applyTheme() {
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeController.swift]: that would make sense. I try to come up with a better solution

File: CriticalMass/ChatNavigationButton.swift [hunk 17]
```diff
@@ -8,6 +8,14 @@
 import UIKit
 
 class ChatNavigationButton: CustomButton {
+    @objc
+    dynamic var unreadMessagesBackgroundColor: UIColor? {
+        willSet {
+            unreadLabel.backgroundColor = newValue
+            unreadLabel.layer.backgroundColor = newValue?.cgColor
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ChatNavigationButton.swift]: setting the layer is unneeded. will remove that

File: CriticalMass/AppDelegate.swift [hunk 18]
```diff
@@ -35,6 +35,10 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+
+        ThemeController().applyTheme()
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/AppDelegate.swift]: Since it is more like a DI Container I didn't think of that. Where would you put it?

File: CriticalMass/ThemeDefining.swift [hunk 19]
```diff
@@ -0,0 +1,161 @@
...
+        return true
+    }
+
+    var placeholderTextColor: UIColor {
+        return .gray300
+    }
+}
+
+struct DarkTheme: ThemeDefining {
+    var name: String {
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeDefining.swift]: I removed it before. Did you check out the latest code?

File: CriticalMass/ThemeDefining.swift [hunk 20]
```diff
@@ -0,0 +1,161 @@
...
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 10.04.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+protocol ThemeDefining {
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeDefining.swift]: think it works already:
Screenshot 2019-05-01 at 21 15 07
Screenshot 2019-05-01 at 21 15 15

File: CriticalMass/ChatInputView.swift [hunk 21]
```diff
@@ -12,37 +12,63 @@ protocol ChatInputDelegate: AnyObject {
...
+    dynamic var textViewTextColor: UIColor? {
+        willSet {
+            textField.textColor = newValue
+        }
+    }
+
+    @objc
+    dynamic var seperatorColor: UIColor? {
+        willSet {
+            separator.backgroundColor = newValue
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ChatInputView.swift]: true. There ist actually no seperator, right?

File: CriticalMass/ChatInputView.swift [hunk 22]
```diff
@@ -12,37 +12,63 @@ protocol ChatInputDelegate: AnyObject {
 }
 
 class ChatInputView: UIView, UITextFieldDelegate {
+    @objc
+    dynamic var chatInputBackgroundColor: UIColor? {
+        willSet {
+            backgroundColor = newValue
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ChatInputView.swift]: commited unintentionally. thanks

File: CriticalMass/NavigationOverlayViewController.swift [hunk 23]
```diff
@@ -7,26 +7,33 @@
...
 
     let representation: Representation
     let action: Action
 }
 
 class NavigationOverlayViewController: UIViewController {
+    // UIAppearance
+    @objc dynamic var backgroundColor: UIColor? {
+        didSet {
+            view.backgroundColor = backgroundColor
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/NavigationOverlayViewController.swift]: commited unintentionally. thanks

File: CriticalMass/ChatNavigationButton.swift (hunk 3 above)
Comment (mltbnz) [reply CriticalMass/ChatNavigationButton.swift]: fixed that. check out latest changes

File: CriticalMass/AppController.swift (hunk 1 above)
Comment (mltbnz) [reply CriticalMass/AppController.swift]: Makes applying a theme easier since it is only one kind. For using the app with voice over it also makes sense. Plus shortens code

File: CriticalMass/NSNotification+Static.swift [hunk 24]
```diff
@@ -0,0 +1,13 @@
...
+//  NSNotification+Static.swift
+//  CriticalMaps
+//
+//  Created by Malte Bünz on 19.04.19.
+//  Copyright © 2019 Pokus Labs. All rights reserved.
+//
+
+import Foundation
+
+extension NSNotification {
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/NSNotification+Static.swift]: Maybe the other Notifications then should also be updated from NSNotification to Notification. I was adopting to the codebase here. Could also be a future PR that adopts the rest of the Notifications to this extension so we have strings only in this one place

File: CriticalMass/SettingsSection.swift [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+
+extension Hashable where Self: CaseIterable {
+    var index: Self.AllCases.Index {
+        return type(of: self).allCases.firstIndex(of: self)!
+    }
+}
+
+enum Section: Int, CaseIterable {
+    case gps
+    case darkMode
```
Comment (lennet) [inline CriticalMass/SettingsSection.swift]: In my opinion, darkmode and gps should be a model of the preference section instead of representing there own section.

File: CriticalMass/ThemeDefining.swift (hunk 13 above)
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeDefining.swift]: I moved the Color creation in the extension and cleaned the extension up 🧹

File: CriticalMass/UIApplication+Appearance.swift [hunk 26]
```diff
@@ -0,0 +1,34 @@
...
+        for window in windows {
+            window.refreshAppearance(animated: animated)
+        }
+    }
+
+    /// Refreshes appearance for all windows in the application
+    ///
+    /// - Parameter animated: if the refresh should be animated
+    func refreshAppearance(animated: Bool) {
+        NotificationCenter.default.post(name: .CMAppearanceWillRefreshApplication, object: self)
```
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/UIApplication+Appearance.swift]: Thats true. It was intended to have full feature so someone could observe a change and add a custom transition for example. We can think about deleting it. Does not hurt to have them though

File: CriticalMass/SettingsSection.swift (hunk 25 above)
Comment (mltbnz) [reply CriticalMass/SettingsSection.swift]: I updated it to be in preferences

File: CriticalMass/AppDelegate.swift (hunk 18 above)
[the remark this answers is not in the document]
Comment (lennet) [reply CriticalMass/AppDelegate.swift]: You could introduce something like appcontroller.onLaunch()

File: CriticalMass/ThemeDefining.swift (hunk 20 above)
[the remark this answers is not in the document]
Comment (lennet) [reply CriticalMass/ThemeDefining.swift]: Bildschirmfoto 2019-05-03 um 18 44 11

File: CriticalMass/TextFieldWithInsets.swift (hunk 8 above)
Comment (lennet) [reply CriticalMass/TextFieldWithInsets.swift]: yes it would be nice to add comments why this hack is needed

File: CriticalMass/UIWindow+Appearance.swift (hunk 10 above)
Comment (lennet) [reply CriticalMass/UIWindow+Appearance.swift]: To be honest this seems a bit scary. bu it seems like everyone is doing it. We should also add a comment here.

File: CriticalMass/en.lproj/Localizable.strings [hunk 27]
```diff
@@ -140,3 +140,11 @@
 /* No comment provided by engineer. */
 "close.button.label" = "close";
 
+/* Dark Mode */
+"settings.theme.dark" = "Dark Mode";
+
+/* Light Mode */
+"settings.theme.light" = "Light Mode";
```
Comment (lennet) [inline CriticalMass/en.lproj/Localizable.strings]: We don't need that key anymore

File: CriticalMass/en.lproj/Localizable.strings [hunk 28]
```diff
@@ -140,3 +140,11 @@
 /* No comment provided by engineer. */
 "close.button.label" = "close";
```
Comment (lennet) [inline CriticalMass/en.lproj/Localizable.strings]: Italian translation is missing

File: CriticalMass/SettingsViewController.swift [hunk 29]
```diff
@@ -97,12 +56,34 @@ class SettingsViewController: UITableViewController {
...
-        return Section.allCases[section].secionTitle
+    override func tableView(_: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let cell = tableView.dequeueReusableHeaderFooterView(withIdentifier: SettingsTableSectionHeader.typeName)
+        let header = cell as! SettingsTableSectionHeader
+        header.titleLabel.text = Section.allCases[section].secionTitle
+        return cell
+    }
+
+    override func tableView(_: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        guard section == Section.info.index else {
```
Comment (lennet) [inline CriticalMass/SettingsViewController.swift]: Do we still need this addition after moving gps and night mode to the same section?

File: CriticalMass/AppController.swift (hunk 1 above)
Comment (lennet) [reply CriticalMass/AppController.swift]: It doesn't make a difference for VoiceOver and Theming because both solution are using UIButtons in the end. I don't like that we know have UIButton instances in the AppController

File: CriticalMass/UIWindow+Appearance.swift (hunk 10 above)
Comment (mltbnz) [reply CriticalMass/UIWindow+Appearance.swift]: I had the same feeling :D I'll add it

File: CriticalMass/en.lproj/Localizable.strings (hunk 28 above)
Comment (mltbnz) [reply CriticalMass/en.lproj/Localizable.strings]: I'll add a google translate one

File: CriticalMass/SettingsViewController.swift (hunk 29 above)
Comment (mltbnz) [reply CriticalMass/SettingsViewController.swift]: yes. otherwise it adds an empty one to the other sections or actually it is there anyways but only has a height that makes it visible for the info section

File: CriticalMass/ThemeDefining.swift (hunk 20 above)
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/ThemeDefining.swift]: Good point. I didn't even saw it there

File: CriticalMass/AppDelegate.swift (hunk 18 above)
[the remark this answers is not in the document]
Comment (mltbnz) [reply CriticalMass/AppDelegate.swift]: I'll apply it in the init

File: CriticalMass/AppController.swift (hunk 1 above)
Comment (mltbnz) [reply CriticalMass/AppController.swift]: I updated to PR to roll back to the previous solution

**Comment by zutrinken** [comment]
one last thing: The notification bubble number has to be white - in dark and light mode

Bildschirmfoto 2019-05-07 um 14 36 03

**Comment by mltbnz** [comment]
That must have slipped back somewhere. I fixed it and added two unit tests for safety 👍

**Comment by mltbnz** [comment]
💪
